federatedscope/core/auxiliaries/runner_builder.py

```diff
@@ -18,21 +23,37 @@ def get_runner(data, server_class, client_class, config, client_configs=None):
 
     Note:
       The key-value pairs of built-in runner and source are shown below:
-        ===============================  ==============================
-        Mode                             Source
-        ===============================  ==============================
-        ``standalone``                   ``core.fed_runner.StandaloneRunner``
-        ``distributed``                  ``core.fed_runner.DistributedRunner``
-        ===============================  ==============================
+        =============================  ===============================
+        Mode                                          Source
+        =============================  ===============================
+        ``standalone``                 ``core.fed_runner.StandaloneRunner``
+        ``distributed``                ``core.fed_runner.DistributedRunner``
+        ``standalone(process_num>1)``  ``core.auxiliaries.parallel_runner.``
+                                       ``StandaloneMultiGPURunner``
+        =============================  ===============================
     """
 
     mode = config.federate.mode.lower()
+    process_num = config.federate.process_num
 
     if mode == 'standalone':
-        runner_cls = StandaloneRunner
+        if process_num <= 1:
+            runner_cls = StandaloneRunner
+        else:
+            runner_cls = StandaloneMultiGPURunner
     elif mode == 'distributed':
         runner_cls = DistributedRunner
 
+    # federated dataset might change the number of clients
+    # thus, we allow the creation procedure of dataset to modify the global
+    # cfg object
+    if runner_cls is not StandaloneMultiGPURunner:
```

Comment:
Maybe we can change this branch to `if runner_cls is StandaloneMultiGPURunner, then data=None`